docs/source/guide/pec-4-low-level.myst

```diff
@@ -0,0 +1,224 @@
+
++++
+
+## Step 1: Probabilistic generation of all the auxiliary circuits
+
++++
+
+### Define the circuit of interest and the quasi-probability representations
+
++++
+
+We define the circuit of interest as we did in Section [How do I use PEC?](pec-1-intro.myst)
+
+```{code-cell} ipython3
+import mitiq
+from mitiq import benchmarks
+
+frontend = "qiskit"  # Supported: "cirq", "qiskit", "pyquil", "braket", "pennylane".
+
+circuit = benchmarks.generate_rb_circuits(
+  n_qubits=1, num_cliffords=2, return_type = frontend,
+)[0]
+print(circuit)
+```
+
+We assume local depolarizing noise and we define the list of {class}`.OperationRepresentation`s (one for each gate)
+as we did in Section [How do I use PEC?](pec-1-intro.myst).
+
+```{code-cell} ipython3
+from mitiq.pec.representations.depolarizing import represent_operations_in_circuit_with_local_depolarizing_noise
+
+noise_level = 0.01
+reps = represent_operations_in_circuit_with_local_depolarizing_noise(circuit, noise_level)
+print(f"{len(reps)} OperationRepresentation objects produced, assuming {noise_level :.2%} depolarizing noise.")
+```
+
+Each element of `reps` is an {class}`.OperationRepresentation` object that represents an ideal gate
+$\mathcal G_i$ of `circuit` as a linear combination of noisy operations $\{\mathcal O_{i, \alpha} \}$:
+
+$$ \mathcal G_i = \sum_\alpha \eta_{i, \alpha} \mathcal O_{i, \alpha}, \quad \sum_\alpha \eta_{i, \alpha}
+= 1, \quad \sum_\alpha |\eta_{i,\alpha}|=\gamma_i,$$
+
+where $\{\eta_{i, \alpha}\}$ is a real quasi-probability distribution with one-norm $\gamma_i$.
+
+For example, the first {class}`.OperationRepresentation` in reps is:
+
+```{code-cell} ipython3
+print(reps[0])
+```
+
+### Probabilistic generation of all the auxiliary circuits
+
++++
+
+According to the [theory of PEC](pec-5-theory.myst), to probabilistically generate an auxiliary circuit one should:
+- Define an empty `sampled_circuit` to be populated with probabilistic operations $\mathcal O_{j,\alpha}$;
+- Define an empty `gate_sign_list` to be populated with the sign values ${\rm sgn}(\eta_{j,\alpha})$;
+- Start a loop over the ideal operations of `circuit`;
+- For each gate of `circuit`, search for the corresponding {class}`.OperationRepresentation` in `reps`;
+- Sample a noisy gate from the quasi-probability distribution of the ideal gate using `rep[j].sample()` as
+   shown in [What additional options are available in PEC?](pec-3-options.myst);
+- Append the sampled gate to `sampled_circuit` and the corresponding sign to `gate_sign_list`;
+- When the loop is completed, return `sampled_circuit` and the associated `sampled_sign` (which is the product of all
+   the elements of `gate_sign_list`).
+
+Instead of manually applying all the previous steps, one can call the Mitiq function {func}`.sample_circuit()` to probabilistically generate an integer number `num_samples` of auxiliary circuits.
+
+```{code-cell} ipython3
+from mitiq import pec
+
+pec.sample_circuit(circuit, reps, num_samples=5)
```

Comment:
Would be nice to see the circuits in this output, but really that's a Qiskit problem.

![image](https://user-images.githubusercontent.com/32416820/146085284-f614954c-752b-46e1-ae3f-196ce8d896fa.png)

Reply:
I agree. Done here: a326948 .
I picked an _ad hoc_ seed such that we don't sample 5 equal circuits. In this way it is more clear how PEC circuits may look like.